### PR TITLE
Record dynacast requirement of a subscriber synchronously.

### PR DESCRIPTION
**Pull request by boks1971**
With rapid changes to subscription settings, use of a goroutine
could end up processing dynacast needs for that subscriber in
a different order. So, record the susbcription needs of a subscriber
in the callback and process the data in a go routine.

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -973,9 +973,10 @@ func (p *ParticipantImpl) UpdateSubscribedTrackSettings(trackID livekit.TrackID,
 
 	subTrack := p.subscribedTracks[trackID]
 	if subTrack == nil {
+		// will get set when subscribed track is added
 		p.lock.Unlock()
-		p.params.Logger.Warnw("could not find subscribed track", nil, "trackID", trackID)
-		return errors.New("could not find subscribed track")
+		p.params.Logger.Infow("could not find subscribed track", nil, "trackID", trackID)
+		return nil
```
Comment (boks1971) [inline pkg/rtc/participant.go:979]: @cnderrauber also made this change, will resolve once that is merged.

File: pkg/rtc/participant.go [hunk 2]
```diff
@@ -1969,6 +1970,7 @@ func (p *ParticipantImpl) rtcpSendWorker() {
 	// read from rtcpChan
 	for pkts := range p.rtcpCh {
 		if pkts == nil {
+			p.params.Logger.Infow("exiting RTCP send worker")
```
Comment (boks1971) [inline pkg/rtc/participant.go:1973]: Seeing what I think is a case of RTCP worker getting shut down unexpectedly. So, logging.

File: pkg/sfu/buffer/rtpstats.go [hunk 3]
```diff
@@ -271,6 +267,10 @@ func (r *RTPStats) Update(rtph *rtp.Header, payloadSize int, paddingSize int, pa
 			r.bytesPadding += pktSize
 		} else {
 			r.bytes += pktSize
+
+			if rtph.Marker {
+				r.frames++
+			}
```
Comment (boks1971) [inline pkg/sfu/buffer/rtpstats.go:273]: Moving the frame counter to only non-duplicate path.

**Review by davidzhao — APPROVED** [review]
lg

**Review by cnderrauber — APPROVED** [review]

**Pull request merged**